src/sim/UI/ViewClass.py

```diff
@@ -157,6 +211,34 @@ def __yTextboxHandler(self, text):
             self.__yTextboxVal = 0
 
 
+    def __addBlockerButtonClicked(self):
+        self.__controller.addNewBlocker(self.__blockerVals[0], self.__blockerVals[1], self.__blockerVals[2], self.__blockerVals[3])
+
+    def __xBTextboxHandler(self, text):
+        try:
+            self.__blockerVals[0] = int(text)
+        except:
+            self.__blockerVals[0] = 0
+
+    def __yBTextboxHandler(self, text):
+        try:
+            self.__blockerVals[1] = int(text)
+        except:
+            self.__blockerVals[1] = 0
+    
+    def __widthTextboxHandler(self, text):
```

Comment:
Width should only be allowed POSITIVE integers:

When I entered negative integers, I couldn't move the blocker around or enter any subsequent blockers.